doc/coq.rst

```diff
@@ -0,0 +1,383 @@
+used in the current project. See the :ref:`Dune Coq language section<coq-lang>`
+for more details.
+
+The Coq theory stanza is very similar in form to the OCaml :ref:`library`
+stanza:
+
+.. code:: scheme
+
+    (coq.theory
+     (name <module_prefix>)
+     (package <package>)
+     (synopsis <text>)
+     (modules <ordered_set_lang>)
+     (libraries <ocaml_libraries>)
+     (flags <coq_flags>)
+     (mode <coq_native_mode>)
+     (theories <coq_theories>))
+
+The stanza will build all ``.v`` files in the given directory and also all
+subdirs if the :ref:`include-subdirs <include-subdirs-coq>` stanza is present.
+The semantics of fields are:
+
+- ``<module_prefix>`` is a dot-separated list of valid Coq module names and
+  determines the module scope under which the theory is compiled (this
+  corresponds to Coq's ``-R`` option).
+  
+  For example, if ``<module_prefix>`` is ``foo.Bar``, the theory modules will be
+  named as ``foo.Bar.module1``, ``foo.Bar.module2``, etc. Note that modules in
+  the same theory don't see the ``foo.Bar`` prefix in the same way that OCaml
+  ``wrapped`` libraries do.
+  
+  For compatibility reasons, Coq lang 1.0 will install a theory named
+  ``foo.Bar`` under ``foo/Bar``. Also note that Coq supports composing a module
+  path from different theories, thus you can name a theory ``foo.Bar`` and a
+  second one ``foo.Baz``, and dune will compose these properly.
+
+- The ``modules`` field allows one to constrain the set of modules included in
+  the theory, similarly to its OCaml counterpart. Modules are specified in Coq
+  notation. That is to say, ``A/b.v`` is written ``A.b`` in this field.
+
+- If ``package`` is present, Dune will generate install rules for the ``.vo``
+  files of the theory. ``pkg_name`` must be a valid package name.
+  
+  Note that Coq lang 1.0 will use the Coq legacy install setup, where all
+  packages share a common root namespace and install directory,
+  ``lib/coq/user-contrib/<module_prefix>``, as customary in the Make-based Coq
+  package ecosystem.
+  
+  For compatibility, we also install under the ``user-contrib`` prefix the
+  ``.cmxs`` files that appear in ``<ocaml_libraries>``,
+
+- ``<coq_flags>`` will be passed to ``coqc`` as command-line options.
+  ``:standard`` is taken from the value set in the ``(coq (flags <flags>))``
+  field in ``env`` profile. See :ref:`dune-env` for more information.
+
+- The path to the ``<ocaml_libraries>`` installed locations will be passed to
+  ``coqdep`` and ``coqc`` using Coq's ``-I`` flag; this allows for a Coq theory
+  to depend on an ML library, making it a Coq plugin.
+
+- Your Coq theory can depend on other theories by specifying them in the
+  ``<coq_theories>`` field. Dune will then pass to Coq the corresponding flags
+  for everything to compile correctly (this corresponds to the ``-Q`` flag for
+  Coq).
+  
+  As of today, we only support composition with libraries defined in the same
```

Comment:
Coq users won't know exactly what you mean by "composition".  They shouldn't have to search through this doc to find out what it means--they don't want to read any more of this doc than they have to (it's a reference manual).